service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -116,12 +164,11 @@ public void onConnectionBound(Event event)
                 ((TransportInternal)transport).addTransportLayer(webSocket);
             }
 
-            Sasl sasl = transport.sasl();
-            sasl.plain(this.userName, this.sasToken);
+            transport.sasl().setMechanisms("ANONYMOUS");
```

Comment:
Note that this does not mean the connection will have no authentication. It means we will defer authentication to the CBS links

Reply:
can you elaborate a bit? what was the impact when the mechanism was missing? is Anonymous the default mechanism?

Reply:
Can you add this same explanation as a comment in code?

Reply:
> Can you add this same explanation as a comment in code?

Sure thing

Reply:
> can you elaborate a bit? what was the impact when the mechanism was missing? 

This mechanism was never missing, it was just set to "PLAIN" by the sasl.plain(); call that I deleted below.

> is Anonymous the default mechanism?

I'm not sure that AMQP has a "default mechanism"

Reply:
+1 for adding comment. :)